### PR TITLE
PR for Auth

### PR DESCRIPTION
**Pull request by suryxks**
features implemented:

-    Login
-    Logout
-    signup

test credentials:
email:  adarshbalika@gmail.com
password:  adarshBalika123

[preview](https://h0j98j.csb.app/)

### CODE REVIEW

**Review by sakshamAK — APPROVED** [review]
Nice work! The code is really good and clean. I liked how you have arranged everything in a new line when it starts getting longer which made it easy to read in the import statements.

Just a suggestion, at some points the code got a bit difficult to read like in home.jsx file from line 9 to line 24. 

So you can do something like, putting line 17 to line 19 in one line, giving space after every function like after line 14 and line 26.

File: src/frontend/Pages/Signin/Signin.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { Navbar } from "../../components";
+import { Link, useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { useAuth } from "../../contexts/AuthContext.jsx";
+import axios from "axios";
+
+export const Signin = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
```
Comment (poornima-alugubelly) [inline src/frontend/Pages/Signin/Signin.jsx]: u can make email and password into a single state called formValues

File: src/frontend/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -16,6 +19,19 @@ const Navbar = () => {
...
         <SearchBar />
         <div className="actions">
+          <button
+            className="btn-cta text-center"
+            onClick={() => {
+              if (isAuthenticated()) {
+                logout();
+                navigate("/signin");
+              } else {
+                navigate("/signin");
```
Comment (poornima-alugubelly) [inline src/frontend/components/Navbar/Navbar.jsx]: instead of writing navigate two times here u can just write it once after ifelse

**Review by poornima-alugubelly — APPROVED** [review]

**Comment by suryxks** [comment]
> Nice work! The code is really good and clean. I liked how you have arranged everything in a new line when it starts getting longer which made it easy to read in the import statements.
> 
> Just a suggestion, at some points the code got a bit difficult to read like in home.jsx file from line 9 to line 24.
> 
> So you can do something like, putting line 17 to line 19 in one line, giving space after every function like after line 14 and line 26.

Thanks saksham ,it was done by  code formatter will look into it

**Pull request merged**